modules/kafka/src/main/scala/higherkindness/mu/kafka/ConsumerStream.scala

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2017-2020 47 Degrees, LLC. <http://www.47deg.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package higherkindness.mu.kafka
+
+import cats.effect.{ConcurrentEffect, ContextShift, Sync, Timer}
+import fs2.Stream
+import fs2.kafka.{ConsumerSettings, KafkaConsumer}
+import higherkindness.mu.format.Decoder
+import io.chrisdavenport.log4cats.Logger
+import io.chrisdavenport.log4cats.slf4j.Slf4jLogger
+
+object ConsumerStream {
+
+  def apply[F[_]: Sync, A](topic: String, settings: ConsumerSettings[F, String, Array[Byte]])(
+      implicit contextShift: ContextShift[F],
+      concurrentEffect: ConcurrentEffect[F],
+      timer: Timer[F],
+      decoder: Decoder[A],
+      sync: Sync[F]
+  ): Stream[F, A] =
+    for {
+      implicit0(logger: Logger[F]) <- fs2.Stream.eval(Slf4jLogger.create[F])
+      s                            <- apply(fs2.kafka.consumerStream(settings))(topic)
+    } yield s
+
+  private[kafka] def apply[F[_], A](
+      kafkaConsumerStream: Stream[F, KafkaConsumer[F, String, Array[Byte]]]
+  )(topic: String)(
+      implicit contextShift: ContextShift[F],
+      concurrentEffect: ConcurrentEffect[F],
+      timer: Timer[F],
+      decoder: Decoder[A],
+      logger: Logger[F]
+  ): Stream[F, A] =
+    kafkaConsumerStream
+      .evalTap(_.subscribeTo(topic))
+      .flatMap(
+        _.stream
+          .flatMap { message =>
+            val a = decoder.decode(message.record.value)
+            for {
+              _ <- fs2.Stream.eval(logger.info(a.toString))
+            } yield a
```

Comment:
where do you commit the read messages? 🤔

Reply:
@tzimisce012 With h.m.k.consumerSetting.atLeastOnceFromEarliest setttings, I am leaving it to be auto.

Reply:
Oh, I just saw this `.withEnableAutoCommit(true)`

Reply:
I think we'll need to iterate on this later. Right now I only see `atLeastOnceFromEarliest`, which turns on auto-commit. We'll probably want to provide more flexibility in commit strategies, possibly by exposing fs2-kafka's `CommitableConsumerRecord` or something equivalent to it.

Reply:
Yes, at the moment it is quite rigid. I wanted to avoid getting this PR longer than it is now 😄. So I added only a simple case. 

I want to develop the example project further, `mu-kafka-sandbox` by adding more complicated scenarios to feed into the `mu-kafka` development.